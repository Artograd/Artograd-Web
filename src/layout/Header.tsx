import {
  BurgerButton,
  Button,
  MainMenu,
  FlexSpacer,
  FlexCell,
  Burger,
  FlexRow,
  MainMenuButton,
} from '@epam/uui';
import { AdaptiveItemProps, MainMenuLogo } from '@epam/uui-components';
import { useLocation, useHistory } from 'react-router-dom';
import styles from './Header.module.scss';
import { useTranslation } from 'react-i18next';

import { RootState } from '../store/store';
import { useSelector } from 'react-redux';
import { Avatar } from './components/Avatar/Avatar';
import { LanguageSelector } from './components/LanguageSelector/LanguageSelector';

const cognitoLoginUrl = `${
  process.env.REACT_APP_LOGIN_URL
}&redirect_uri=${encodeURIComponent(process.env.REACT_APP_REDIRECT_URL ?? '')}`;

const cognitoSignUpUrl = `${
  process.env.REACT_APP_REGISTER_URL
}&redirect_uri=${encodeURIComponent(process.env.REACT_APP_REDIRECT_URL ?? '')}`;

export const Header = ({ mobile = false }: { mobile?: boolean }) => {
  const location = useLocation();
  const history = useHistory();

  const { t } = useTranslation();

  const { isLoggedIn } = useSelector((state: RootState) => state.identity);

  const visitPage = (props: { onClose: () => void }, page: string) => {
    props.onClose;
    props.onClose();
    history.push(page);
  };

  const renderBurger = (props: { onClose: () => void }) => (
    <>
      <BurgerButton
        caption={t('global.layout.header.homepage')}
        onClick={() => visitPage(props, '/')}
      />
      <BurgerButton
        caption={t('global.layout.header.tenders')}
        onClick={() => visitPage(props, '/tenders')}
      />
      <BurgerButton
        caption={t('global.layout.header.proposals')}
        onClick={() => visitPage(props, '/proposals')}
      />
      {!isLoggedIn && (
        <BurgerButton
          caption={t('global.layout.header.signInCta')}
          href={cognitoLoginUrl}
        />
      )}
      {!isLoggedIn && (
        <BurgerButton
          caption={t('global.layout.header.signUpCta')}
          onClick={() => visitPage(props, '/register')}
        />
      )}
    </>
  );

  const getAnonMenuItems = (): AdaptiveItemProps[] => {
    return [
      {
        id: 'burger',
        priority: 100,
        collapsedContainer: !mobile,
        render: (p) => (
          <Burger
            key={p.id}
            width={300}
            renderBurgerContent={renderBurger}
            rawProps={{ 'data-testid': `header-burger-menu` }}
          />
        ),
      },
      {
        id: 'logo',
        priority: 99,
        render: (p) => (
          <MainMenuLogo
            key={p.id}
            onClick={() => history.push('/')}
            logoUrl="artograd.logo.svg"
          />
        ),
      },
      {
        id: 'homeMenuItem',
        priority: 1,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            onClick={() => history.push('/')}
            caption={t('global.layout.header.homepage')}
            isLinkActive={location.pathname === '/'}
            cx={styles.menuPageLink}
          />
        ),
      },
      {
        id: 'tendersMenuItem',
        priority: 2,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            onClick={() => history.push('/tenders')}
            caption={t('global.layout.header.tenders')}
            isLinkActive={location.pathname === '/tenders'}
            cx={styles.menuPageLink}
          />
        ),
      },
      {
        id: 'proposalsMenuItem',
        priority: 3,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            onClick={() => history.push('/proposals')}
            caption={t('global.layout.header.proposals')}
            isLinkActive={location.pathname === '/proposals'}
            cx={styles.menuPageLink}
          />
        ),
      },
      {
        id: 'flexSpacer2',
        priority: 100,
        render: (p) => <FlexSpacer key={p.id} />,
      },
      {
        id: 'languageSelector',
        priority: 7,
        render: () => <LanguageSelector />,
      },
      {
        id: 'signin',
        priority: 5,
        render: (p) => (
          <FlexRow key={p.id} padding="6" vPadding="12" spacing="12">
            <Button
              key={p.id}
              href={cognitoLoginUrl}
              caption={t('global.layout.header.signInCta')}
              fill="none"
              color="primary"
              cx={styles.signInButton}
            />
          </FlexRow>
        ),
      },
      {
        id: 'signup',
        priority: 6,
        render: (p) => (
          <FlexRow key={p.id} padding="6" vPadding="12" spacing="12">
            <Button
              href={cognitoSignUpUrl}
              color="primary"
              caption={t('global.layout.header.signUpCta')}
              cx={styles.signUpButton}
            />
          </FlexRow>
        ),
      },
    ];
  };

  const getLoggedInMenuItems = (): AdaptiveItemProps[] => {
    return [
      {
        id: 'burger',
        priority: 100,
        collapsedContainer: !mobile,
        render: (p) => (
          <Burger
            key={p.id}
            width={300}
            renderBurgerContent={renderBurger}
            rawProps={{ 'data-testid': `header-burger-menu` }}
          />
        ),
      },
      {
        id: 'logo',
        priority: 99,
        render: (p) => (
          <MainMenuLogo
            key={p.id}
            onClick={() => history.push('/')}
            logoUrl="artograd.logo.svg"
          />
        ),
      },
      {
        id: 'homeMenuItem',
        priority: 1,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            onClick={() => history.push('/')}
            caption={t('global.layout.header.homepage')}
            isLinkActive={location.pathname === '/'}
            cx={styles.menuPageLink}
          />
        ),
      },
      {
        id: 'tendersMenuItem',
        priority: 2,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            onClick={() => history.push('/tenders')}
            caption={t('global.layout.header.tenders')}
            isLinkActive={location.pathname === '/tenders'}
            cx={styles.menuPageLink}
          />
        ),
      },
      {
        id: 'proposalsMenuItem',
        priority: 3,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            onClick={() => history.push('/proposals')}
            caption={t('global.layout.header.proposals')}
            isLinkActive={location.pathname === '/proposals'}
            cx={styles.menuPageLink}
          />
        ),
      },
      {
        id: 'flexSpacer2',
        priority: 100,
        render: (p) => <FlexSpacer key={p.id} />,
      },
      {
        id: 'languageSelector',
        priority: 7,
        render: () => <LanguageSelector />,
      },
      {
        id: 'header-avatar',
        priority: 5,
        render: () => <Avatar />,
      },
    ];
  };

  return (
    <FlexCell>
      <MainMenu
        items={isLoggedIn ? getLoggedInMenuItems() : getAnonMenuItems()}
      />
    </FlexCell>
  );
};
