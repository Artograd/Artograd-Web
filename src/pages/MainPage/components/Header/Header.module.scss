@use '~@epam/promo/assets/styles/fonts' as *;

.headerWrapper {
  background: #d9e2fc;
  padding: 0px 120px;
}

.title {
  font-size: 44px;
  padding-top: 60px;
  margin: 0;
  color: #111111;
  font-family: var(--font-museo-sans);
  font-weight: 600;
  line-height: 50px;
}

.subtitle {
  font-size: 24px;
  line-height: 30px;
  margin: 0;
  padding-top: 24px;
  color: #474a59;
  white-space: pre-line;
  font-family: var(--font-museo-sans);
  font-weight: 600;
}

.cta {
  margin: 40px 0 60px 0;
}
