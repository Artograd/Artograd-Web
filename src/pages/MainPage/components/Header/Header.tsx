import { FlexRow, Button, Text } from '@epam/uui';
import { FlexCell } from '@epam/uui-components';
import styles from './Header.module.scss';

export const Header = () => {
  return (
    <FlexRow>
      <FlexCell width="100%" cx={styles.headerWrapper}>
        <FlexRow>
          <Text cx={styles.title}>Welcome to Artograd!</Text>
        </FlexRow>
        <FlexRow>
          <Text cx={styles.subtitle}>
            The largest community of creators. <br />
            Be the one who changes the world around.
          </Text>
        </FlexRow>
        <FlexRow>
          <Button
            cx={styles.cta}
            color="primary"
            caption="Join Creators Community"
            onClick={() => null}
          />
        </FlexRow>
      </FlexCell>
      {/* TODO: ADD PICTURE */}
    </FlexRow>
  );
};
