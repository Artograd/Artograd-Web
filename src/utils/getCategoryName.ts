import { CategoryItemType } from '../types';

export const categoryList: CategoryItemType[] = [
  { id: 0, name: 'tendersPage.newTender.categories.sculptures' },
  { id: 1, name: 'tendersPage.newTender.categories.mosaics' },
  { id: 2, name: 'tendersPage.newTender.categories.murals' },
  { id: 3, name: 'tendersPage.newTender.categories.graffiti' },
  { id: 4, name: 'tendersPage.newTender.categories.functionalArt' },
  {
    id: 5,
    name: 'tendersPage.newTender.categories.interactiveInstallations',
  },
  { id: 6, name: 'tendersPage.newTender.categories.botanicalArt' },
  { id: 7, name: 'tendersPage.newTender.categories.waterFeatures' },
  { id: 8, name: 'tendersPage.newTender.categories.themedGardens' },
  { id: 9, name: 'tendersPage.newTender.categories.recycled' },
];

export const getCategoryName = (id: string) => {
  return categoryList.find((category) => category.id === Number(id));
};
