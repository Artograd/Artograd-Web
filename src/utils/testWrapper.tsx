import { renderWithContextAsync } from '@epam/uui-test-utils';
import { MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { store } from '../store/store';

export const testWrapper = async ({
  component,
  history,
}: {
  component: React.JSX.Element;
  history: MemoryHistory<unknown>;
  
}) => {
  return renderWithContextAsync(<Provider store={store}><Router history={history}>{component}</Router></Provider>);
};
