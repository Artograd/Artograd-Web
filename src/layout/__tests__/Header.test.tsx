import { Header } from '../Header';
import { createMemoryHistory } from 'history';
import { screen, fireEvent } from '@epam/uui-test-utils';
import { testWrapper } from '../../utils/testWrapper';
import { initialState } from '../../utils/testWrapper';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/test',
  }),
}));

const menuLinks = [
  {
    linkName: 'Home',
    url: '/',
  },
  {
    linkName: 'Tenders',
    url: '/tenders',
  },
  {
    linkName: 'Art Objects',
    url: '/artobjects',
  },
];

describe('Layout mobile header', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('renders correctly', async () => {
    const history = createMemoryHistory();

    const component = await testWrapper({
      component: <Header mobile />,
      history,
    });
    fireEvent.click(screen.getByTestId('header-burger-menu'));

    expect(component).toMatchSnapshot();
  });

  test('redirect on burger menu link click', async () => {
    const history = createMemoryHistory();

    await testWrapper({ component: <Header mobile />, history });

    fireEvent.click(screen.getByTestId('header-burger-menu'));
    fireEvent.click(screen.getByText(menuLinks[1].linkName));

    expect(history.location.pathname).toBe(menuLinks[1].url);
  });
});

describe('Layout header', () => {
  const history = createMemoryHistory();
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('renders correctly', async () => {
    const component = await testWrapper({ component: <Header />, history });

    expect(component).toMatchSnapshot();
  });

  menuLinks.map((link) => {
    test(`[ ${link.linkName} ] redirect on menu link click`, async () => {
      await testWrapper({ component: <Header />, history });
      fireEvent.click(screen.getByText(link.linkName));

      expect(history.location.pathname).toBe(link.url);
    });
  });

  menuLinks.map((link) => {
    test(`[ ${link.linkName} ]redirect on menu link click with loggedIn user`, async () => {
      await testWrapper({
        component: <Header />,
        history,
        state: {
          ...initialState,
          identity: {
            ...initialState.identity,
            isLoggedIn: true,
          },
        },
      });
      fireEvent.click(screen.getByText(link.linkName));

      expect(history.location.pathname).toBe(link.url);
    });
  });

  test('change language on click', async () => {
    await testWrapper({ component: <Header />, history });
    fireEvent.click(screen.getByText('en'));
    await fireEvent.click(screen.getByText('Русский'));

    expect(screen.getByText('Главная'));
  });

  test('goes back to home page on logo click with logged in user', async () => {
    const history = createMemoryHistory();

    await testWrapper({
      component: <Header />,
      history,
      state: {
        ...initialState,
        identity: {
          ...initialState.identity,
          isLoggedIn: true,
        },
      },
    });

    fireEvent.click(screen.getByTestId('header-logo-image'));

    expect(history.location.pathname).toBe('/');
  });

  test('goes back to home page on logo click anon user', async () => {
    const history = createMemoryHistory();

    await testWrapper({
      component: <Header />,
      history,
    });

    fireEvent.click(screen.getByTestId('header-logo-image'));

    expect(history.location.pathname).toBe('/');
  });
});
