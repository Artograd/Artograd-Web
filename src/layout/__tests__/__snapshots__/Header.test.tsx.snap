// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout header renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
        style="min-width: 0; flex-basis: 0px;"
      >
        <nav
          class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
        >
          <div
            class="QFvWHT rnyczA"
          >
            <div
              class="uui-flex-row TtA74w _3odmFf"
            >
              <div>
                <a
                  aria-disabled="true"
                  class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                  tabindex="-1"
                >
                  <img
                    alt="Main Menu Logo"
                    class="KnIMMk"
                    src="artograd.logo.svg"
                  />
                </a>
              </div>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Home
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Tenders
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Proposals
                </div>
              </button>
              <div
                class="G9zrcQ"
              />
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- languageSelector"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    en
                  </div>
                  <div
                    class="kb-S-l uui-icon uui-enabled"
                  >
                    <svg
                      class=""
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </button>
              </div>
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                  href="https://artograd.auth.eu-central-1.amazoncognito.com/oauth2/authorize?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign In
                  </div>
                </a>
              </div>
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                  href="https://artograd.auth.eu-central-1.amazoncognito.com/signup?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign Up
                  </div>
                </a>
              </div>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
      style="min-width: 0; flex-basis: 0px;"
    >
      <nav
        class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
      >
        <div
          class="QFvWHT rnyczA"
        >
          <div
            class="uui-flex-row TtA74w _3odmFf"
          >
            <div>
              <a
                aria-disabled="true"
                class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                tabindex="-1"
              >
                <img
                  alt="Main Menu Logo"
                  class="KnIMMk"
                  src="artograd.logo.svg"
                />
              </a>
            </div>
            <button
              class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Home
              </div>
            </button>
            <button
              class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Tenders
              </div>
            </button>
            <button
              class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Proposals
              </div>
            </button>
            <div
              class="G9zrcQ"
            />
            <div
              class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- languageSelector"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  en
                </div>
                <div
                  class="kb-S-l uui-icon uui-enabled"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </button>
            </div>
            <div
              class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
            >
              <a
                class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                href="https://artograd.auth.eu-central-1.amazoncognito.com/oauth2/authorize?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                tabindex="0"
              >
                <div
                  class="uui-caption"
                >
                  Sign In
                </div>
              </a>
            </div>
            <div
              class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
            >
              <a
                class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                href="https://artograd.auth.eu-central-1.amazoncognito.com/signup?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                tabindex="0"
              >
                <div
                  class="uui-caption"
                >
                  Sign Up
                </div>
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Layout mobile header renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
        style="min-width: 0; flex-basis: 0px;"
      >
        <nav
          class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
        >
          <div
            class="QFvWHT rnyczA"
          >
            <div
              class="uui-flex-row TtA74w _3odmFf"
            >
              <div
                class="rvz1XB uui-burger uui-burger-menu _9vPsqZ"
                data-testid="header-burger-menu"
              >
                <button
                  class="uui-burger-button"
                >
                  <div
                    class="kb-S-l uui-icon uui-enabled"
                  >
                    <svg
                      class=""
                      height="60"
                      viewBox="0 0 24 24"
                      width="60"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 9.2h8v.8H8zM8 14h6v.8H8zm0-2.4h8v.8H8z"
                      />
                    </svg>
                  </div>
                </button>
              </div>
              <div>
                <a
                  aria-disabled="true"
                  class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                  tabindex="-1"
                >
                  <img
                    alt="Main Menu Logo"
                    class="KnIMMk"
                    src="artograd.logo.svg"
                  />
                </a>
              </div>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Home
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Tenders
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Proposals
                </div>
              </button>
              <div
                class="G9zrcQ"
              />
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- languageSelector"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    en
                  </div>
                  <div
                    class="kb-S-l uui-icon uui-enabled"
                  >
                    <svg
                      class=""
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </button>
              </div>
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                  href="https://artograd.auth.eu-central-1.amazoncognito.com/oauth2/authorize?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign In
                  </div>
                </a>
              </div>
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                  href="https://artograd.auth.eu-central-1.amazoncognito.com/signup?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign Up
                  </div>
                </a>
              </div>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
      style="min-width: 0; flex-basis: 0px;"
    >
      <nav
        class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
      >
        <div
          class="QFvWHT rnyczA"
        >
          <div
            class="uui-flex-row TtA74w _3odmFf"
          >
            <div
              class="rvz1XB uui-burger uui-burger-menu _9vPsqZ"
              data-testid="header-burger-menu"
            >
              <button
                class="uui-burger-button"
              >
                <div
                  class="kb-S-l uui-icon uui-enabled"
                >
                  <svg
                    class=""
                    height="60"
                    viewBox="0 0 24 24"
                    width="60"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 9.2h8v.8H8zM8 14h6v.8H8zm0-2.4h8v.8H8z"
                    />
                  </svg>
                </div>
              </button>
            </div>
            <div>
              <a
                aria-disabled="true"
                class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                tabindex="-1"
              >
                <img
                  alt="Main Menu Logo"
                  class="KnIMMk"
                  src="artograd.logo.svg"
                />
              </a>
            </div>
            <button
              class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Home
              </div>
            </button>
            <button
              class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Tenders
              </div>
            </button>
            <button
              class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Proposals
              </div>
            </button>
            <div
              class="G9zrcQ"
            />
            <div
              class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- languageSelector"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  en
                </div>
                <div
                  class="kb-S-l uui-icon uui-enabled"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </button>
            </div>
            <div
              class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
            >
              <a
                class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                href="https://artograd.auth.eu-central-1.amazoncognito.com/oauth2/authorize?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                tabindex="0"
              >
                <div
                  class="uui-caption"
                >
                  Sign In
                </div>
              </a>
            </div>
            <div
              class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
            >
              <a
                class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                href="https://artograd.auth.eu-central-1.amazoncognito.com/signup?client_id=16lm3r1hhua3p00unpbojdsi21&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                tabindex="0"
              >
                <div
                  class="uui-caption"
                >
                  Sign Up
                </div>
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
